File: android/src/main/java/com/margelo/nitro/nitrovideo/enums/DRMType.kt
package com.margelo.nitro.nitrovideo.enums

import androidx.media3.common.C
import java.util.UUID

enum class DRMType(val value: String) {
    CLEARKEY("clearkey"),
    FAIRPLAY("fairplay"),
    PLAYREADY("playready"),
    WIDEVINE("widevine");

    fun isSupported(): Boolean {
        return this != FAIRPLAY
    }

    fun toUUID(): UUID {
        return when (this) {
            CLEARKEY -> C.CLEARKEY_UUID
            FAIRPLAY -> throw UnsupportedOperationException("FairPlay DRM is only supported on Apple platforms (iOS/macOS).")
            PLAYREADY -> C.PLAYREADY_UUID
            WIDEVINE -> C.WIDEVINE_UUID
        }
    }

    companion object {
        fun fromString(str: String?): DRMType? {
            return entries.firstOrNull { it.value.equals(str, ignoreCase = true) }
        }
    }
}

File: ios/Enums/DRMType.swift
import Foundation

public enum DRMType: String, Codable, Sendable {
  case clearkey
  case fairplay
  case playready
  case widevine

  public func isSupported() -> Bool {
    return self == .fairplay
  }

  public func assertIsSupported() throws {
    if !isSupported() {
      throw DRMUnsupportedException(self.rawValue)
    }
  }
}
